### ISSUE TITLE
Does not polyfill Promise returned from fetch on Firefox 49

### ISSUE THREAD

**Issue by torarnek** [es-shims/Promise.prototype.finally#4]
Our experience is that the returned Promise from fetch on Firefox is not polyfilled with finally. This works on Chrome and other browsers.

**Comment by ljharb** [comment]
This seems like a Firefox bug with `fetch` (which isn't part of the ES spec) - could you provide a jsfiddle that I could use to reproduce it?

**Comment by torarnek** [comment]
I am sorry, but I have no time to make a jsfiddle now.

**Comment by ljharb** [comment]
`Promise.prototype.finally = 3; window.fetch().finally === 3` works fine for me in Firefox 49 (as do the tests for this module)

Without repro steps, I can't really do much :-/ happy to reopen if we can reproduce it!

**Comment by jackfranklin** [comment]
@ljharb I'm going to try to repro on JSFiddle but I'm seeing similar behaviour on Firefox 50.1:

```js
Promise.prototype.finally = function() { console.log('blah blah') }

window.fetch('foo').finally() => `finally is not a function`
```

I can log `Promise.prototype.finally` and it's fine, but that doesn't seem to make it through to the result of `fetch`.

**Comment by jackfranklin** [comment]
@ljharb I've found the cause of the problem.

TL;DR: Promise.prototype.finally works just fine in Firefox on its own. However, when the core-js library has been used to shim `Promise`, it breaks. 

This is surprising to me as I thought that core-js would just not do anything if it's run in a browser with `Promise` but I guess that's not true. I do have this reproducible if it's of interest, I can push the repo to GitHub.

**Comment by ljharb** [comment]
@jackfranklin i maintain es6-shim, and I heartily recommend that instead :-)

However, I'd prefer to figure out the cause of the issue in `core-js` and file a bug. Could you perhaps make a jsfiddle that reproduces the problem?

**Comment by jonashaag** [comment]
Just a quick heads up. We hit this issue as well. I tried to come up with a reproducing example but I couldn't. Not sure what's going on.

**Comment by ljharb** [comment]
Thanks - as soon as I can repro with core-js + Firefox 49 + this shim, I'll be happy to make a fix.
